src/packages/@ncigdc/utils/withSearch.js

```diff
@@ -52,31 +71,44 @@ export const withSearch = passedInState => {
                 'Content-Type': 'application/json',
               },
             },
-          ).then(response =>
-            handleResults(response.data.query.hits, timeOfRequest),
-          ),
+          ).then(response => {
+            let hits = [];
+            if (response && response.data) {
+              hits = response.data.query.hits;
+            }
+            return handleResults(hits, timeOfRequest, query);
+          }),
         );
       },
     }),
     withHandlers({
       selectItem: ({ push, reset }) => (item: ISearchHit) => {
-        push(
-          `/${atob(item.id)
-            .split(':')[0]
-            .toLocaleLowerCase()}s/${atob(item.id).split(':')[1]}`,
-        );
+        item.uuid
+          ? push(`/files/${item.uuid}`)
```

Comment:
Should we break this out into a "extractId" function rather than this ternary if?

Reply:
👍